### PR TITLE
feat(dashboards): Add Release Health fields to widget modal 

### PR DESCRIPTION
**Pull request by shruthilayaj**
Fitler to Release Health Metrics Fields and tags when the
Metrics dataset selector is chosen. There's a lot of logic
that's required to show the right fields, the QueryFields
& WidgetQueryFields components are getting particularly
bloated, but holding off on any refactors till we move logic
over to the full page widget builder.

Tests will be added to this PR. Opening this without
tests to unblock any future work.

Depends on https://github.com/getsentry/sentry/pull/31660

### CODE REVIEW

File: static/app/views/dashboardsV2/widget/metricWidget/fields.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import {SelectValue} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {FieldValue, FieldValueKind} from 'sentry/views/eventsV2/table/types';
+
+export type MetricsColumnType = 'set' | 'counter';
+
+export enum FieldKey {
+  SESSION = 'session',
+  USER = 'user',
```
Comment (matejminar) [inline static/app/views/dashboardsV2/widget/metricWidget/fields.tsx:9]:
```suggestion
  SESSION = 'sentry.sessions.session',
  USER = 'sentry.sessions.user',
```

File: static/app/views/dashboardsV2/widget/metricWidget/fields.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+export const METRICS_FIELDS: Readonly<Record<FieldKey, MetricsColumnType>> = {
```
Comment (matejminar) [inline static/app/views/dashboardsV2/widget/metricWidget/fields.tsx:12]: We might need to fetch this from API: 
https://www.notion.so/sentry/Metrics-API-373168c15ed44ebbbca300e9a3861949#480c6303852b4c5a9fa9dfe3815363cc
Hardcoding might not be enough, especially with custom metrics.

Not a blocker for release health though 👍

**Review by matejminar — APPROVED** [review]
Looks good to me, left a few minor comments.

File: static/app/views/dashboardsV2/widget/metricWidget/fields.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/dashboardsV2/widget/metricWidget/fields.tsx:9]: should we use the fields defined in this file?

https://github.com/getsentry/sentry/blob/eb5781c217833acf605cbf4ec7de5e39b6756614/static/app/utils/metrics/fields.tsx#L5

File: static/app/views/eventsV2/table/queryField.tsx [hunk 3]
```diff
@@ -628,6 +630,7 @@ function validateColumnTypes(
     return columnTypes({name: input.meta.name, dataType: input.meta.dataType});
   }
 
+  // @ts-ignore
   return columnTypes.includes(input.meta.dataType);
```
[the remark this answers is not in the document]
Comment (shruthilayaj) [reply static/app/views/eventsV2/table/queryField.tsx:634]: haha thank you! I was trying to figure out how to get rid of the ts error but got frustrated and forgot about it

File: static/app/views/eventsV2/table/types.tsx [hunk 4]
```diff
@@ -6,6 +6,7 @@ import {
   ColumnType,
   ColumnValueType,
 } from 'sentry/utils/discover/fields';
+import {MetricsColumnType} from 'sentry/views/dashboardsV2/widget/metricWidget/fields';
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/table/types.tsx:9]:
```suggestion
import {MetricsColumnType} from 'sentry/views/dashboardsV2/widgetBuilder/metricWidget/fields';
```

**Review by priscilawebdev — APPROVED** [review]
we have to fix some imports, but other than that the code looks good to me

**Pull request merged**